src/main/java/rlbot/manager/BotManager.java

```diff
@@ -0,0 +1,86 @@
+package rlbot.manager;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+import rlbot.Bot;
+import rlbot.api.GameData;
+import rlbot.cpp.RLBotDll;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class BotManager {
+
+    private final Map<Integer, BotProcess> botProcesses = new HashMap<>();
+
+    private boolean keepRunning;
+
+    private final Object dinnerBell = new Object();
+    private  GameData.GameTickPacket latestPacket;
+
+    public void registerBot(final int index, final Bot bot) {
+        if (botProcesses.containsKey(index)) {
+            return;
+        }
+
+        botProcesses.computeIfAbsent(index, (idx) -> {
+            Thread botThread = new Thread(() -> doRunBot(bot));
+            botThread.start();
+            return new BotProcess(bot, botThread);
+        });
+    }
+
+    private void doRunBot(Bot bot) {
+        while (keepRunning) {
```

Comment:
instead of a while loop you could try using a timer.
It would allow us to call it more with the 120 hz like we do with the python bots,
I think there is a way to have the timer wait until the previous call has completed.

Reply:
This loop does not need a timer. `dinnerBell.wait(1000);` will cause this thread to wait until a fresh packet is available.

Your comment applies better to the doLoop method where I'm sleeping for 16 ms.  There's room for improvement there, and one solution is 120hz, but I'd like to hold off on that for now because I think we can do better. We could make the dll block until a fresh packet is available, or we could implement your long-lasting callback idea.